File: go-api/src/models/ShippingAddress.go
package models

type ShippingAddress struct {
	ID            int32  `json:"id"`
	CustomerID    int32  `json:"customer_id"`
	FirstName     string `json:"first_name"`
	LastName      string `json:"last_name"`
	Phone         int32  `json:"phone"`
	PostCode      int16  `json:"post_code"`
	City          string `json:"city"`
	State         string `json:"state"`
	Address       string `json:"address"`
	ApartmentSuit string `json:"apartment_suit"`
	Country       string `json:"country"`
	CreatedAt     string `json:"created_at"`
	IsDefault     int8   `json:"is_default"`
}

File: nodejs-api/src/models/ShippingAddress.ts
export type ShippingAddressType = {
  _id?: string
  customer_id: string
  firstName: string
  lastName: string
  phone: number
  post_code: number
  city: string
  state: string
  address: string
  apartment_suit: string
  country: string
  email?: string
  createdAt: Date
  isDefault: boolean
}

import mongoose from "mongoose";
import {ObjectFlags} from "../types";

const schemaObject: ObjectFlags<ShippingAddressType> = {
  address: String,
  apartment_suit: String,
  city: String,
  country: String,
  createdAt: Date,
  customer_id: { type: mongoose.Schema.Types.ObjectId, ref: "User", index: true },
  firstName: String,
  isDefault: { type: Boolean, default: true },
  lastName: String,
  phone: Number,
  post_code: Number,
  state: String,
  email: String
}

mongoose.model(
  'ShippingAddress',
  new mongoose.Schema(schemaObject, {timestamps: true})
);
